### ISSUE TITLE
The pre-delivery gate's "frontmatter parses" check silently turns into a regex when PyYAML is missing, and passes a description wrapped onto an unindented line

### ISSUE THREAD

**Issue by chibuzorokoh** [rebelytics/one-skill-to-rule-them-all#158]
**Target:** `scripts/validate-skill-bundle.py`, `check_dir()`, and gate item (4) in `references/weekly-review.md`, Delivering updated skills.
**Verified at:** upstream `main` (`7518a85`) and the v3.1.0 install (`2967fa5`). The relevant code is identical in both.

## What happened

A cosmetic pass rewrapped the long lines of a staged `SKILL.md` and did not exclude the frontmatter. The single-line `description:` became a key line followed by unindented continuation lines, which is not valid YAML. The gate printed `OK: all gate checks passed`, including item (4), "the frontmatter parses". The bundle installed. The skill body loaded fine, so the defect showed up only in the client's skill listing, where the description had been replaced by the H1 title.

## Reproduction

```bash
mkdir -p demo-skill
printf -- '---\nname: demo-skill\ndescription: Use when a staged skill needs a demonstration of a wrapped description value that\ncontinues on an unindented second line, which is not valid YAML for a plain scalar.\n---\n\n# Demo\n' > demo-skill/SKILL.md
python3 scripts/validate-skill-bundle.py demo-skill; echo "exit=$?"
ruby -ryaml -e 'p YAML.safe_load(File.read("demo-skill/SKILL.md").split(/^---\s*$/)[1])'
```

| Instrument | Verdict |
|---|---|
| `validate-skill-bundle.py` at `2967fa5`, `python3` without PyYAML | `OK: all gate checks passed`, exit 0 |
| `validate-skill-bundle.py` at `7518a85`, same interpreter | `OK: all gate checks passed`, exit 0 |
| Ruby Psych (`YAML.safe_load`) | `could not find expected ':' while scanning a simple key at line 4 column 1` |
| `validate-skill-bundle.py` at `2967fa5`, `python3` with PyYAML 6.0.3 installed | FAIL, frontmatter parse error, exit non-zero |

A correctly formed single-line description passes all three, so the fixture is testing the wrap and nothing else.

## Why the current text did not prevent it

```python
try:
    import yaml  # optional; fall back to regex checks if absent
    data = yaml.safe_load(fm)
    ...
except ImportError:
    data = {"name": ..., "description": folded_description(fm)}
```

- **The fallback cannot fail on this input.** `folded_description()` matches `^description:\s*(.+)$` and takes only the first line. The unindented continuation lines are never looked at, and nothing else in the fallback checks the block's structure.
- **The fallback is silent.** Both modes print the same `OK`. Neither the output nor the exit status says whether item (4) was checked by a parser or by a pattern match, so the gate's report reads as the consumer's verdict in both cases.
- **On a common platform the fallback is the default.** On the machine that hit this, none of the three `python3` interpreters on `PATH` (the system one and the Homebrew ones) had PyYAML. The "optional" branch is the one that actually runs.

This is #28's lesson ("test against the installer's acceptance criteria") undone by an optional import. The check exists, but whether it runs depends on the environment, and nothing reports which way it went.

## Suggested fix

1. **Never report item (4) as passed when no parser ran.** In the `ImportError` branch, either fail item (4) with `frontmatter: not parsed — no YAML loader available (install PyYAML, or re-run with --allow-regex-frontmatter)`, or at minimum print a mandatory `WARN: frontmatter checked by pattern only` and end with a different final line, e.g. `OK (frontmatter NOT parsed)`.
2. **Give the fallback a structural floor.** Inside the frontmatter block, every non-blank line must be a top-level `key:` at column 0, an indented line (block-scalar or nested content), or a list item. An unindented line with no `key:` fails. That catches this defect without a YAML library.
3. **Add a planted-input self-test.** A description wrapped onto an unindented line must fail item (4) in both modes. It is the smallest control that shows the check can fail on the machine where it is running.
4. **Delivery text, one sentence:** any whitespace, wrap or reflow normalisation of a staged `SKILL.md` is limited to the body below the closing `---`, because the frontmatter is the one region where a line break carries meaning. (This fits alongside #126's rule against letting a reflow ride along with a content pass.)

Verified after the fact: with PyYAML 6.0.3 installed on the same machine, the same gate fails the same fixture and still passes a correctly formed skill. So the check is sound when the parser is present; the defect is entirely in what happens when it is not.

## Principle

A gate that checks a property by the shape of the text, rather than by running the consumer's parser, passes exactly the inputs the consumer rejects on form, and reports a green result that reads as the consumer's verdict. When the parser is an optional dependency, whether the gate really checks anything depends on the environment. The cheap way to know which is a planted malformed input that the gate is seen to refuse, on the machine where it runs.

**Comment by rebelytics** [comment]
Shipped in v3.3.0 — frontmatter shape is checked structurally whether or not PyYAML is present. Thanks a lot.
